docs/articles/clustering/member-roles.md

```diff
@@ -0,0 +1,246 @@
+
+# How to Use Roles
+
+The member roles are defined in the configuration property named `akka.cluster.roles` and typically defined in the start script as a system property or environment variable.:
+
+```hocon
+akka
+{
+  cluster
+  {
+    roles = ["backend"]
+  }
+}
+```
+
+## Using Roles Within Your Cluster
+
+The roles are part of the membership information in `MemberEvent` that you can subscribe to. The roles of the local cluster member are available from the `SelfMember` and that can be used for conditionally starting certain actors:
+
+```csharp
+var selfMember = Cluster.Get(_actorSystem).SelfMember;
+if (selfMember.HasRole("backend"))
+{
+    context.ActorOf(Backend.Prop(), "back");
+}
+else if (selfMember.HasRole("front"))
+{
+    context.ActorOf(Frontend.Prop(), "front");
+}
+```
+
+## Akka.Cluster.Sharding
+
+Cluster Sharding uses its own Distributed Data Replicator per node. If using roles with sharding there is one Replicator per role, which enables a subset of all nodes for some entity types and another subset for other entity types. Each replicator has a name that contains the node role and therefore the role configuration must be the same on all nodes in the cluster, for example you can’t change the roles when performing a rolling update. Changing roles requires a full cluster restart.
+
+```hocon
+akka
+{
+  cluster
+  {
+    roles = ["worker", "notifier", "credit", "storage"]
+    sharding
+    {
+      role = "worker"
+    }
+  }
+}
+```
+
+```csharp
+var sharding = ClusterSharding.Get(system);
+var shardRegion = await sharding.StartAsync(
+    typeName: "customer",
+    entityPropsFactory: e => Props.Create(() => new Customer(e)),
+    settings: ClusterShardingSettings.Create(system).WithRole("worker"),
+    messageExtractor: new MessageExtractor(10));
+```
+
+## `DistributedPubSub`
+
+Start the mediator on members tagged with this role. All members are used if undefined or empty.
+
+```hocon
+akka
+{
+  cluster
+  {
+    roles = ["worker", "notifier", "credit", "storage"]
+    pub-sub
+    {
+      role = "notifier"
+    }
+  }
+}
+```
+
+## `DData`
+
+Replicas are running on members tagged with this role. All members are used if undefined or empty
+
+```hocon
+akka
+{
+  cluster
+  {
+    roles = ["worker", "notifier", "credit", "storage"]
+    distributed-data
+    {
+      role = "storage"
+    }
+  }
+}
+```
+
+## `ClusterSingleton`
+
+Singleton among the nodes tagged with specified role. If the role is not specified it's a singleton among all nodes in the cluster.
+
+```hocon
+akka
+{
+  cluster
+  {
+    roles = ["worker", "notifier", "credit", "storage"]
+    singleton 
+    {
+      role = "credit"
+    }
+  }
+}
+```
+
+## Cluster-Aware Router
+
+The major benefit of Akka.Cluster is that you can scale out your actor system to more nodes as load on the system increases - in other words, during peak period, Akka.Cluster can scale out your, for instance, order processor. 
```

Comment:
You need to talk about this primarily in the context of roles here, not just the cluster